nni/retiarii/hub/pytorch/autoformer.py

```diff
@@ -267,87 +277,141 @@ class AutoformerSpace(nn.Module):
+        num_classes: int = 1000,
+        qkv_bias: bool = False,
+        drop_rate: float = 0.,
+        attn_drop_rate: float = 0.,
+        drop_path_rate: float = 0.,
+        pre_norm: bool = True,
+        global_pool: bool = False,
+        abs_pos: bool = True,
+        qk_scale: Optional[float] = None,
+        rpe: bool = True,
     ):
         super().__init__()
-
+        # define search space parameters
         embed_dim = nn.ValueChoice(list(search_embed_dim), label="embed_dim")
-        fixed_embed_dim = nn.ModelParameterChoice(
-            list(search_embed_dim), label="embed_dim")
         depth = nn.ValueChoice(list(search_depth), label="depth")
+        mlp_ratios = [nn.ValueChoice(list(search_mlp_ratio), label=f"mlp_ratio_{i}") for i in range(max(search_depth))]
+        num_heads = [nn.ValueChoice(list(search_num_heads), label=f"num_head_{i}") for i in range(max(search_depth))]
+
         self.patch_embed = nn.Conv2d(
-            in_chans,
-            cast(int, embed_dim),
-            kernel_size=patch_size,
-            stride=patch_size)
+            in_chans, cast(int, embed_dim),
+            kernel_size = patch_size,
+            stride = patch_size
+        )
         self.patches_num = int((img_size // patch_size) ** 2)
         self.global_pool = global_pool
-        self.cls_token = nn.Parameter(torch.zeros(1, 1, cast(int, fixed_embed_dim)))
-        trunc_normal_(self.cls_token, std=.02)
 
-        dpr = [
-            x.item() for x in torch.linspace(
-                0,
-                drop_path_rate,
-                max(search_depth))]  # stochastic depth decay rule
-
-        self.abs_pos = abs_pos
-        if self.abs_pos:
-            self.pos_embed = nn.Parameter(torch.zeros(
-                1, self.patches_num + 1, cast(int, fixed_embed_dim)))
-            trunc_normal_(self.pos_embed, std=.02)
-
-        self.blocks = nn.Repeat(lambda index: nn.LayerChoice([
-            TransformerEncoderLayer(embed_dim=embed_dim,
-                                    fixed_embed_dim=fixed_embed_dim,
-                                    num_heads=num_heads, mlp_ratio=mlp_ratio,
-                                    qkv_bias=qkv_bias, drop_rate=drop_rate,
-                                    attn_drop=attn_drop_rate,
-                                    drop_path=dpr[index],
-                                    rpe_length=img_size // patch_size,
-                                    qk_scale=qk_scale, rpe=rpe,
-                                    pre_norm=pre_norm,)
-            for mlp_ratio, num_heads in itertools.product(search_mlp_ratio, search_num_heads)
-        ], label=f'layer{index}'), depth)
-        self.pre_norm = pre_norm
-        if self.pre_norm:
-            self.norm = nn.LayerNorm(cast(int, embed_dim))
-        self.head = nn.Linear(
-            cast(int, embed_dim),
-            num_classes) if num_classes > 0 else nn.Identity()
+        self.cls_token = ClsToken(max(search_embed_dim))
```

Comment:
For the fixed architecture, `embed_dim` is a fixed constant. I think there is no need to use `max` here.